src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -278,6 +272,74 @@ private Stage createBuildStage(ProgrammingLanguage programmingLanguage, ProjectT
         }
     }
 
+    /**
+     * Adds the Bamboo final task for the static code analysis to the
+     * @param defaultJob job to which the task should be added
+     * @param isMavenProject whether the project is a Maven build (or implicitly a Gradle build)
+     */
+    private void setStaticCodeAnalysisJobsForJavaAndKotlinExercise(Job defaultJob, boolean isMavenProject) {
+        // Create artifacts and a final task for the execution of static code analysis
+        List<StaticCodeAnalysisTool> staticCodeAnalysisTools = StaticCodeAnalysisTool.getToolsForProgrammingLanguage(ProgrammingLanguage.JAVA);
+        String command = StaticCodeAnalysisTool.createBuildPlanCommandForProgrammingLanguage(ProgrammingLanguage.JAVA);
+        Artifact[] artifacts = staticCodeAnalysisTools.stream()
+                .map(tool -> new Artifact().name(tool.getArtifactLabel()).location("target").copyPattern(tool.getFilePattern()).shared(false)).toArray(Artifact[]::new);
+        if (isMavenProject) {
+            defaultJob.finalTasks(new MavenTask().goal(command).jdk("JDK").executableLabel("Maven 3").description("Static Code Analysis").hasTests(false));
+        }
+        else {
+            defaultJob.finalTasks(new ScriptTask().inlineBody("./gradlew check").description("Static Code Analysis"));
+        }
+        defaultJob.artifacts(artifacts);
+    }
+
+    /**
+     * Adds the tasks for executing a non-sequential test run to the stage and returns the stage for the Bamboo Build Plan for Java and Kotlin Exercises.
+     * @param defaultStage the stage to which the jobs are added
+     * @param defaultJob the job to which the tasks are added
+     * @param checkoutTask the vcs checkout task
+     * @param isMavenProject whether the project is a Maven project (or implicitly a Gradle project)
+     * @return the stage with the tasks for the non-sequential test run
+     */
+    private Stage getNonSequentialTestTaskForJavaAndKotlinExercise(Stage defaultStage, Job defaultJob, VcsCheckoutTask checkoutTask, boolean isMavenProject) {
```

Comment:
```suggestion
    private Stage getRegularTestTaskForJavaAndKotlinExercise(Stage defaultStage, Job defaultJob, VcsCheckoutTask checkoutTask, boolean isMavenProject) {
```
I think they are usually called regular test runs instead of non-sequential.